src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java

```diff
@@ -39,64 +43,71 @@
+        if (whatHappened == WhatHappened.initialized || SECRETS_TOPIC.equals(node.getName())
+                || PERIODIC_REFRESH_INTERVAL_MIN.equals(node.getName())) {
             serviceChanged();
         }
     };
-
     @Inject
     SecretManagerIPCAgent secretManagerIPCAgent;
-
     @Inject
     private GreengrassCoreIPCService greengrassCoreIPCService;
+    // for testing
+    @Setter
+    private CountDownLatch isInitialSyncComplete = new CountDownLatch(1);
 
     /**
      * Constructor for SecretManagerService Service.
-     * @param topics                root Configuration topic for this service
-     * @param secretManager         secret manager which manages secrets
-     * @param authorizationHandler  authorization handler
-     * @param executorService       executor service
+     *
+     * @param topics               root Configuration topic for this service
+     * @param secretManager        secret manager which manages secrets
+     * @param authorizationHandler authorization handler
+     * @param executorService executor service
+     * @param ses scheduled executor service
      */
     @Inject
-    public SecretManagerService(Topics topics,
-                                SecretManager secretManager,
-                                AuthorizationHandler authorizationHandler,
-                                ExecutorService executorService) {
+    public SecretManagerService(Topics topics, SecretManager secretManager, AuthorizationHandler authorizationHandler,
+                                ExecutorService executorService, ScheduledExecutorService ses) {
         super(topics);
         this.secretManager = secretManager;
         this.authorizationHandler = authorizationHandler;
         this.executor = executorService;
+        this.ses = ses;
     }
 
     @Override
     protected void install() throws InterruptedException {
         super.install();
+        // Re-initialize the sync counter every time the component is installed
+        setIsInitialSyncComplete(new CountDownLatch(1));
         // subscribe will invoke serviceChanged right away to sync from cloud
         this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this.handleConfigurationChangeLambda);
     }
 
     private void serviceChanged() {
-        replaceSyncFuture(secretManager::syncFromCloud);
-    }
-
-    private synchronized Future<?> replaceSyncFuture(Runnable r) {
-        Future<?> newFut = r == null ? null : executor.submit(r);
-        Future<?> oldFut = syncFuture.getAndSet(newFut);
-        if (oldFut != null) {
-            oldFut.cancel(true);
+        long refreshIntervalSeconds = (long) (Coerce.toDouble(this.config.lookupTopics(CONFIGURATION_CONFIG_KEY)
+                .findOrDefault(60, PERIODIC_REFRESH_INTERVAL_MIN)) * 60);
+        ScheduledFuture<?> oldScheduledSyncFut = scheduledSyncFuture.getAndSet(ses.scheduleAtFixedRate(() -> {
+            secretManager.syncFromCloud();
```

Comment:
lets let this throw and catch the exception here so we don't count down isInitialSyncComplete in failure case